Update EIP-1: Add `finalized` date to the preamble (#5517)
ethereum/EIPs: EIPS/eip-1.md

`created`: *Date the EIP was created on*

> `finalized`: *Date the EIP's status turns FINAL* (Optional field, only needed when turning FINAL)

Reviewer: How this:

```suggestion
`finalized`: *Date the EIP was finalized* (Optional field, only needed when status is `Final` or `Withdrawn`)
```

Author: I hesitate to use "finalized" for the EIPs of withdrawn, I wonder if it shall have a separate date or just a more general date of "last updated"

Reviewer: In EIP-1's description of `Withdrawn`:

> This state has finality and can no longer be resurrected using this EIP number.

I think `finalized` applies to both, since they have "finality".

Author: That makes sense
